### PR TITLE
chore(forms-tw):add type test e2e test

### PR DESCRIPTION
**Pull request by chinook25**
### What are the main changes you did
- closes https://github.com/molgenis/GCC/issues/2774

### How to test
- The e2e tests should succeed.

### Checklist
- [ ] updated docs in case of new feature
- [x] added/updated tests
- [ ] added/updated testplan to include a test for this fix, including ref to bug using # notation

By creating this pull request I have agreed with the [contributor terms](https://github.com/molgenis/molgenis-emx2/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: apps/tailwind-components/playwright.config.ts [hunk 1]
```diff
@@ -42,6 +42,7 @@ export default defineConfig<ConfigOptions>({
       name: "chromium",
       use: { ...devices["Desktop Chrome"] },
       testIgnore: "*/admin!*.spec.ts",
+      testMatch: "*/*.spec.ts",
```
Comment (chinook25) [inline apps/tailwind-components/playwright.config.ts:45]: Needed to get the vscode plugin to see the tests.

File: apps/tailwind-components/tests/e2e/components/form/add-type-test.spec.ts [hunk 2]
```diff
@@ -0,0 +1,627 @@
...
+    .getByRole("textbox", { name: "string type Required" })
+    .fill("some string");
+
+  // String array
+  await page
+    .locator('[id="type test-Types-stringArrayType-form-field-input_0"]')
+    .click();
+  await page
+    .locator('[id="type test-Types-stringArrayType-form-field-input_0"]')
+    .fill("some string in an array");
```
Comment (davidruvolo51) [inline apps/tailwind-components/tests/e2e/components/form/add-type-test.spec.ts]: There seems to be an issue with the generation of the input IDs as there is a space in the ID. This would might make the selector unstable. It would be good to sanitise the IDs to make sure they follow the same format. Probably better to do this in a new PR

File: apps/tailwind-components/tests/e2e/components/form/add-type-test.spec.ts [hunk 3]
```diff
@@ -0,0 +1,627 @@
...
+  await page.goto(route + "form/AddModal.story?schema=type+test&table=Types");
+  await page.getByRole("button", { name: "Add Types" }).click();
+
+  // File and Multiselect tests are not working yet
+
+  // String
+  await page.getByRole("textbox", { name: "string type Required" }).click();
+  await page
+    .getByRole("textbox", { name: "string type Required" })
+    .fill("some string");
```
Comment (davidruvolo51) [inline apps/tailwind-components/tests/e2e/components/form/add-type-test.spec.ts]: I don't think the `click` is needed. `getByRole` and `fill` should be enough to enter values.

File: apps/tailwind-components/tests/e2e/components/form/add-type-test.spec.ts
Comment (davidruvolo51) [inline apps/tailwind-components/tests/e2e/components/form/add-type-test.spec.ts:1]: This is a good test. I think it would be better to split these up into smaller units so it would be easier to see which one failed.

```ts
test.describe("Type test", () => {

    test("String input", () => {
        await page.getByRole("textbox", { name: "string type Required" }).fill("some string");
    });

    test("String array", () => {
          ....
    });
})
```

File: apps/tailwind-components/tests/e2e/components/form/add-type-test.spec.ts [hunk 4]
```diff
@@ -0,0 +1,627 @@
...
+    .locator('[id="type test-Types-decimalArrayType-form-field-input_0"]')
+    .click();
+  await page
+    .locator('[id="type test-Types-decimalArrayType-form-field-input_0"]')
+    .fill("2.2");
+  await page
+    .locator(
+      '[id="type test-Types-decimalArrayType-form-field"] > div:nth-child(3) > .flex.items-center.justify-center'
+    )
+    .click();
```
Comment (davidruvolo51) [inline apps/tailwind-components/tests/e2e/components/form/add-type-test.spec.ts:126]: If possible, it would better to avoid using descendant selectors and tailwind classes. Is there an option to select by a role or ID?

**Review by davidruvolo51 — COMMENTED** [review]
Great test! This is something that we need. I have a few comments (see below).

I think some of the comments are issues that can be fixed in a separate PR (e.g., input IDs, missing roles, missing IDs, etc.). I would focus on-

- Organisation of the tests (we do this in some of the other e2e tests)
- Check: is `click` needed? (I think this automatically generated by the playwright record. If that's what was used)

File: apps/tailwind-components/tests/e2e/components/form/add-type-test.spec.ts (hunk 2 above)
Comment (chinook25) [reply apps/tailwind-components/tests/e2e/components/form/add-type-test.spec.ts]: Would be good yes, but agree on that being out of scope.

File: apps/tailwind-components/tests/e2e/components/form/add-type-test.spec.ts (hunk 3 above)
Comment (chinook25) [reply apps/tailwind-components/tests/e2e/components/form/add-type-test.spec.ts]: great, removed a lot of them (I think the rest is still needed).

File: apps/tailwind-components/tests/e2e/components/form/add-type-test.spec.ts (hunk 4 above)
Comment (chinook25) [reply apps/tailwind-components/tests/e2e/components/form/add-type-test.spec.ts:126]: not in all cases, but rewrote a lot of them :)

File: apps/tailwind-components/tests/e2e/components/form/add-type-test.spec.ts
Comment (chinook25) [reply apps/tailwind-components/tests/e2e/components/form/add-type-test.spec.ts:1]: I'm unsure of this one, since the point of the test was to have all fields filled and tested in one go. Splitting it might be good as additional tests that also go more into edge cases, however that would also be a lot slower. I'd like to leave that for a future extension of the test.

Comment (davidruvolo51) [reply apps/tailwind-components/tests/e2e/components/form/add-type-test.spec.ts:1]: Good point. Let's leave it for now.

**Review by davidruvolo51 — APPROVED** [review]
Looks good!
